src/Popup/index.js

```diff
@@ -150,15 +153,19 @@ class Popup extends Component {
     }
 
     reposition() {
-        if (this.props.visible) {
-            // TODO(@narqo): don't call DOMNode measurements in case nothing has changed
-            const layout = this.calcBestLayoutParams();
+        if (!this.props.visible) {
+            return;
+        }
 
-            if (this.props.onLayout) {
-                this.props.onLayout({ layout }, this.props);
-            }
+        // TODO(@narqo): don't call DOMNode measurements in case nothing has changed
+        const layout = this.calcBestLayoutParams();
+
+        if (this.props.onLayout) {
+            this.props.onLayout({ layout }, this.props);
+        }
 
-            const { direction, left, top } = layout;
+        const { direction, left, top } = layout;
+        if (this.state.direction !== direction || this.state.left !== left || this.state.top !== top) {
```

Comment:
The problem is not how the comparison is made, but in the way how this peace is implemented. We have to call `reposition` in `componentDidMount`, as it must have an access to the re-rendered DOM, but, at the same time, we have to call `setState` here, as we must update CSS-classes and styles (and I don't want to do it directly in DOM). This overall leads to inefficient code and lots of hacks :(